src/Queue/QueueBase.php

```diff
@@ -84,12 +84,11 @@ public function __construct($name, Connection $connection,
     $this->connection = $connection;
     $this->logger = $logger;
     $this->modules = $modules;
-
     // Declare any exchanges required if configured.
     $exchanges = $config->get('exchanges');
     if ($exchanges) {
       foreach ($exchanges as $name => $exchange) {
-        $this->getChannel()->exchange_declare(
+        $this->connection->getConnection()->channel()->exchange_declare(
```

Comment:
yes we can do that if the exchange already exist, but if the exchange doesn't exist you'll get exchange not_found error....

Anyway according to the docs, the workflow first we should declare exchange.